nginx.conf

```diff
@@ -0,0 +1,21 @@
+events {
+    worker_connections 1024;
+}
+
+http {
+    upstream backend {
+        server backend:3000;  
+    }
+
+    server {
+        listen 80;
+
+        location / {
+            proxy_pass http://backend;
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
```

Comment:
why this is used?

Reply:
this ensures the reverse proxy that the backend server has access to the client's original IP address and the protocol used, which helps maintain accurate client information and proper handling of the requests. i have put it from watching tutorial on youtube but removing these lines wont affect the overall code